test/integration/testDataType.js

```diff
@@ -216,7 +217,7 @@ describe('Test DataType', function ()
 
   describe('testSemiStructuredDataType', function ()
   {
-    it('testVariant', function (done)
+    it('testVariantJSON', function (done)
```

Comment:
can we add additional describe and tests where we override default json and xml parser?

alternative implementation for JSON could be just `JSON.parse` and for xml we can leave todo to implement the test in https://github.com/snowflakedb/snowflake-connector-nodejs/pull/538 (e.g. `TODO SNOW-830291: add custom xml parser test` should be enough)

Reply:
I tried using `JSON.parse()` but it failed because the test data contained `Infinity` and `undefined` which doesn't seem to be supported by JSON

I made another insert query with only valid JSON and it works